### PR TITLE
Update GC.xml

### PR DESCRIPTION
**Pull request by jkotas**
## Summary

Add guidelines for when it is profitable to use AddMemoryPressure / RemoveMemoryPressure APIs.

Fixes https://github.com/dotnet/runtime/issues/33812

### CODE REVIEW

File: xml/System/GC.xml [hunk 1]
```diff
@@ -157,9 +157,12 @@
...
+
+ In the simplest usage pattern, a managed object allocates unmanaged memory in the constructor and releases it in the `Finalize` method. Call the <xref:System.GC.AddMemoryPressure%2A> method after allocating the unmanaged memory, and call the <xref:System.GC.RemoveMemoryPressure%2A> method after releasing it.  
   
- In more complicated scenarios, where the unmanaged memory allocation changes substantially during the lifetime of the managed object, you can call the <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.RemoveMemoryPressure%2A> methods to communicate these incremental changes to the runtime.  
+ In more complicated scenarios, where the unmanaged memory allocation changes substantially during the lifetime of the managed object, you can call the <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.
+ssure%2A> methods to communicate these incremental changes to the runtime.
```
Comment (jkotas) [inline xml/System/GC.xml:165]:
```suggestion
 In more complicated scenarios, where the unmanaged memory allocation changes substantially during the lifetime of the managed object, you can call the <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.RemoveMemoryPressure%2A> methods to communicate these incremental changes to the runtime.  
```

File: xml/System/GC.xml [hunk 2]
```diff
@@ -157,7 +157,9 @@
...
   
- In the simplest usage pattern, a managed object allocates unmanaged memory in the constructor and releases it in the `Dispose` or `Finalize` method. Call the <xref:System.GC.AddMemoryPressure%2A> method after allocating the unmanaged memory, and call the <xref:System.GC.RemoveMemoryPressure%2A> method after releasing it.  
+ The <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.ReleaseMemoryPressure%2A> methods improve performance only for types that exclusively depend on finalizers to release the unmanaged resources. It is not necessary to use these methods in types that follow dispose pattern and finalizers are used to cleanup unmanaged resources only in the event that a consumer of the type forgets to call `Dispose` method. For more information on object finalization and the dispose pattern, see [Cleaning Up Unmanaged Resources](/dotnet/standard/garbage-collection/unmanaged).
```
Comment (jkotas) [inline xml/System/GC.xml:160]:
```suggestion
 The <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.RemoveMemoryPressure%2A> methods improve performance only for types that exclusively depend on finalizers to release the unmanaged resources. It is not necessary to use these methods in types that follow dispose pattern and finalizers are used to cleanup unmanaged resources only in the event that a consumer of the type forgets to call `Dispose` method. For more information on object finalization and the dispose pattern, see [Cleaning Up Unmanaged Resources](/dotnet/standard/garbage-collection/unmanaged).
```

File: xml/System/GC.xml [hunk 3]
```diff
@@ -1448,8 +1450,10 @@ This method is most useful in monitoring scenarios for measuring the difference
...
-  
- In the simplest usage pattern, a managed object allocates unmanaged memory in the constructor and releases it in the `Dispose` or `Finalize` method. Call the <xref:System.GC.AddMemoryPressure%2A> method after allocating the unmanaged memory, and call the <xref:System.GC.RemoveMemoryPressure%2A> method after releasing it.  
+
+ The <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.ReleaseMemoryPressure%2A> methods improve performance only for types that exclusively depend on finalizers to release the unmanaged resources. It is not necessary to use these methods in types that follow dispose pattern and finalizers are used to cleanup unmanaged resources only in the event that a consumer of the type forgets to call `Dispose` method. For more information on object finalization and the dispose pattern, see [Cleaning Up Unmanaged Resources](/dotnet/standard/garbage-collection/unmanaged).
```
Comment (jkotas) [inline xml/System/GC.xml:1454]:
```suggestion
 The <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.RemoveMemoryPressure%2A> methods improve performance only for types that exclusively depend on finalizers to release the unmanaged resources. It is not necessary to use these methods in types that follow dispose pattern and finalizers are used to cleanup unmanaged resources only in the event that a consumer of the type forgets to call `Dispose` method. For more information on object finalization and the dispose pattern, see [Cleaning Up Unmanaged Resources](/dotnet/standard/garbage-collection/unmanaged).
```

**Review by Maoni0 — APPROVED** [review]
thanks @jkotas!

File: xml/System/GC.xml [hunk 4]
```diff
@@ -157,7 +157,9 @@
...
   
- In the simplest usage pattern, a managed object allocates unmanaged memory in the constructor and releases it in the `Dispose` or `Finalize` method. Call the <xref:System.GC.AddMemoryPressure%2A> method after allocating the unmanaged memory, and call the <xref:System.GC.RemoveMemoryPressure%2A> method after releasing it.  
+ The <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.RemoveMemoryPressure%2A> methods improve performance only for types that exclusively depend on finalizers to release the unmanaged resources. It is not necessary to use these methods in types that follow dispose pattern and finalizers are used to cleanup unmanaged resources only in the event that a consumer of the type forgets to call `Dispose` method. For more information on object finalization and the dispose pattern, see [Cleaning Up Unmanaged Resources](/dotnet/standard/garbage-collection/unmanaged).
```
Comment (gewarren) [inline xml/System/GC.xml:160]:
```suggestion
 The <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.RemoveMemoryPressure%2A> methods improve performance only for types that exclusively depend on finalizers to release the unmanaged resources. It's not necessary to use these methods in types that follow the dispose pattern, where finalizers are used to clean up unmanaged resources only in the event that a consumer of the type forgets to call `Dispose`. For more information on object finalization and the dispose pattern, see [Cleaning Up Unmanaged Resources](/dotnet/standard/garbage-collection/unmanaged).
```

File: xml/System/GC.xml [hunk 5]
```diff
@@ -1448,8 +1450,10 @@ This method is most useful in monitoring scenarios for measuring the difference
...
-  
- In the simplest usage pattern, a managed object allocates unmanaged memory in the constructor and releases it in the `Dispose` or `Finalize` method. Call the <xref:System.GC.AddMemoryPressure%2A> method after allocating the unmanaged memory, and call the <xref:System.GC.RemoveMemoryPressure%2A> method after releasing it.  
+
+ The <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.RemoveMemoryPressure%2A> methods improve performance only for types that exclusively depend on finalizers to release the unmanaged resources. It is not necessary to use these methods in types that follow dispose pattern and finalizers are used to cleanup unmanaged resources only in the event that a consumer of the type forgets to call `Dispose` method. For more information on object finalization and the dispose pattern, see [Cleaning Up Unmanaged Resources](/dotnet/standard/garbage-collection/unmanaged).
```
Comment (gewarren) [inline xml/System/GC.xml:1454]:
```suggestion
 The <xref:System.GC.AddMemoryPressure%2A> and <xref:System.GC.RemoveMemoryPressure%2A> methods improve performance only for types that exclusively depend on finalizers to release the unmanaged resources. It's not necessary to use these methods in types that follow the dispose pattern, where finalizers are used to clean up unmanaged resources only in the event that a consumer of the type forgets to call `Dispose`. For more information on object finalization and the dispose pattern, see [Cleaning Up Unmanaged Resources](/dotnet/standard/garbage-collection/unmanaged).
```

**Pull request merged**